src/main/scala/chisel3/util/circt/PlusArgsValue.scala

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{annotate, ChiselAnnotation, ExtModule, FlatIO}
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** Create a module which generates a verilog $plusargs$value.  This returns a
+  * value as indicated by the format string and a flag for whether the value
+  * was found.
+  */
+private class PlusArgsValueIntrinsic[T <: Data](gen: T, str: String) extends ExtModule(Map("FORMAT" -> str)) {
+  val io = FlatIO(new Bundle {
+    val found = Output(UInt(1.W))
+    val result = Output(gen)
+  })
+  annotate(new ChiselAnnotation {
+    override def toFirrtl =
+      Intrinsic(toTarget, "circt.plusargs.value")
+  })
+  override val desiredName = "PlusArgsValue_" + Builder.idGen.next
+}
+
+object PlusArgsValue {
+
+  /** Creates an intrinsic which calls $test$plusargs.
+    *
+    * @example {{{
+    * b := PlusArgsValue(UInt<32.W>, "FOO=%d")
+    * b.found
+    * b.value
+    * }}}
+    */
+  def apply[T <: Data](gen: T, str: String) = {
```

Comment:
so is the intrinsic handling the unpacking into some complicated data type from the `%d` which is going to just be an integer? Is this useful vs making that happen explicitly in chisel and make this intrinsic just return a UInt?

Reply:
The intrinsic returns any verilog type which can be parsed with format strings (and is supported from firrtl).  The format string can be used in verilog to pick apart complex specification strings for subfields you are interested in.

Reply:
The intrinsic is nothing more than a wrapper for verilog-spec functionality.

Reply:
right, I am echoing Schuyler's comment above that I'm unclear why this can return anything other than a UInt at this level. Users can easily cast from UInt to their appropriate data type in a safer way than this happening automatically?

Reply:
Now that I understand that this is meaningful in verilog, I am OK with it. We might want to add something to the scaladoc that in terms of order / unpacking it's equivalent to using a UInt and then .asTypeOf(gen).